packages/mobile/src/geth/saga.ts

```diff
@@ -120,7 +128,13 @@ export function* initGethSaga() {
     Logger.error(TAG, 'Geth initialization failed, restarting the app.')
     deleteChainDataAndRestartApp()
   } else {
-    navigateToError('networkConnectionFailed')
+    // Suggest switch to forno for network-related errors
+    if (yield select(promptZeroSyncIfNeededSelector)) {
+      yield put(setPromptZeroSync(false))
+      navigate(Screens.DataSaver, { promptModalVisible: true })
```

Comment:
I assumed we would just enable forno silently in this case. @nityas can you confirm that navigating the user to the Data Saver screen is the desired behavior?

Reply:
The user gets navigated to the DataSaver screen with a modal, then once they confirm/decline the switch they go back to wherever they were

Reply:
Confirmed with @nityas that this prompt is the expected behavior!

Reply:
1. confirmed-- it adds a layer of centralization and we do not want to do this without user consent